### ISSUE TITLE
Add 3D flight editor

### ISSUE THREAD

**Issue by SebastianViollaz** [mavlink/qgroundcontrol#11202]
Adding a 3d editor would be very useful  if you have a properly designed 3d MAP. If you are in an indoor environment using a 3D edit would be easier, and will help avoiding collisions during the flight planification. Maybe adding  some arrows to move the point like 3D design apps:
![image](https://github.com/mavlink/qgroundcontrol/assets/85691148/af3dcd3a-2635-4a67-99a4-c065b12069d5)

**Comment by omid-esrafilian** [comment]
@SebastianViollaz 
Thanks for the suggestion. But do you mean to add an editor to create/edit a map inside the QGC 3D View?

In general, yes that is a good feature, however, it requires extensive work.

**Comment by SebastianViollaz** [comment]
Yes, it's a big feature but would be nice maybe someone is interested in developing it

**Comment by omid-esrafilian** [comment]
But it is not clear to me why someone needs to edit the map inside the QGC?
The map can be for example created using a professional 3D editor software and then imported to the QGC. Does that make sense?

What in my opinion would make more sense is to generalize the current 3D Viewer to also accept and import other file formats such as STL.

**Comment by SebastianViollaz** [comment]
Maybe you misunderstand me, the feature that i propose is to edit that flight path, in the 3d View, not the 3D models of the view.
In this 3D editor you only be able to edit the items positions (Checkpoints,RIOS's, Take Off, RTL etc. 
At the moment you edit in the 2D view, but editing in the 3D view could be nice to determinate the exact position of the points and the exact altitude without having to change to the 3D view to check if the point is in the correct position.


Basically the feature is not a 3D map edit, instead is a 3D flight editor, like the flight editor that QGC already have, but in 3D

**Comment by SebastianViollaz** [comment]
> Adding a 3d editor would be very useful if you have a properly designed 3d MAP. If you are in an indoor environment using a 3D edit would be easier, and will help avoiding collisions during the flight planification. Maybe adding some arrows to move the point like 3D design apps: ![image](https://private-user-images.githubusercontent.com/85691148/310531848-af3dcd3a-2635-4a67-99a4-c065b12069d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MTMwMzMsIm5iZiI6MTcwOTgxMjczMywicGF0aCI6Ii84NTY5MTE0OC8zMTA1MzE4NDgtYWYzZGNkM2EtMjYzNS00YTY3LTk5YTQtYzA2NWIxMjA2OWQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDExNTg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5N2NiZmViYzJhMjUxYzZkODFiYzVhNmUzMzk1NzYxZDkyN2MyOTAwYTRlMTk4MjZiYmIyYmIxYWRhMzQ4MmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AXKvVCV9fSY9yfMTr4pUC5fxyVQBRM-I4RxuBRtT4wU)


The image that i provide here was an idea to how to move the items in the flight plan. Maybe adding arrows to contol de X,Y,Z axis of the elements
.

**Comment by omid-esrafilian** [comment]
Indeed, it would be useful to be able to edit the mission in the 3D view or in 3D in general. I had the same thoughts, as it would be more convenient. 
This could also be considered as an additional feature in the future.
Thanks for the suggestion.
